onnx/checker.cc

```diff
@@ -0,0 +1,247 @@
+#include "onnx/checker.h"
+
+#include "onnx/defs/schema.h"
+#include "onnx/utils.h"
+
+#include <unordered_set>
+
+namespace onnx {
+namespace checker {
+
+#define fail(...) throw ValidationError(onnx::MakeString(__VA_ARGS__));
+
+#define enforce_has_field(proto, field)                                    \
+  if (!proto.has_##field()) {                                              \
+    fail("Field '", #field, "' of ", #proto, " is required but missing."); \
+  }
+
+#define enforce_has_repeated_field(proto, field)                    \
+  if (!proto.field##_size()) {                                      \
+    fail("Repeated Field '", #field, "' is required but missing."); \
+  }
+
+#define enforce_non_empty_field(proto, field)                                  \
+  if (proto.field().empty()) {                                                 \
+    fail("Field '", #field, "' of ", #proto, " is required to be non-empty."); \
+  }
+
+void check_value_info(const ValueInfoProto& value_info, int ir_version) {
+  enforce_non_empty_field(value_info, name);
```

Comment:
Do we have stronger requirements here around names? Not all-whitespace, no embedded nulls/control characters?

Reply:
I think right now the policy is as long as it's non-empty, then it's fine.